Sources/Core/GTMSessionFetcher.m

```diff
@@ -1964,14 +1966,26 @@ - (void)forgetSessionIdentifierForFetcherWithoutSyncCheck {
   }
 }
 
-// External stop method
+// External stop method and don't call callbacks
 - (void)stopFetching {
+  [self stopFetching:NO];
+}
+
+// External stop method with do call callbacks option
+- (void)stopFetching:(BOOL)shouldCallCallbacks {
   @synchronized(self) {
     GTMSessionMonitorSynchronized(self);
 
     // Prevent enqueued callbacks from executing.
     _userStoppedFetching = YES;
   }  // @synchronized(self)
+  if (shouldCallCallbacks) {
+    NSError *error = [NSError errorWithDomain:kGTMSessionFetcherErrorDomain
+                                   code:GTMSessionFetcherErrorCancelled
+                                     userInfo:@{@"description" : @"Operation cancelled"}];
+    _cancelAndCallback = YES;
+    _completionHandler(nil, error);
```

Comment:
The completion handler block should not be invoked directly from the thread calling `stopFetching`, among other things because the completion handler block must be invoked on the `callbackQueue`.

Also, access to the `_completionHandler` ivar needs to be guarded by `@synchronized(self)`.

It would probably be best to add parameter(s) to `stopFetchingReleasingCallbacks:` that triggers invoking the completion handler prior to calling `releaseCallbacks`, following the behavior pattern of `invokeFetchCallbacksOnCallbackQueueWithData:error:mayDecorate:shouldReleaseCallbacks:`, or possibly calls that method directly.